### PR TITLE
chore: bump @types/dockerode version to 3.3.29

### PR DESCRIPTION
**Pull request by axel7083**
### What does this PR do?

https://github.com/DefinitelyTyped/DefinitelyTyped/pull/69385 changed the return type of the `pull` method argument `callback` from `callback: Callback<any>` to `callback: Callback<NodeJS.ReadableStream>`.

The problem are from the `docker-modem` library that was resolving an old version of its types `@types/docker-modem:3.0.2` when only the version `@types/docker-modem:3.0.6` (latest) was compatible with the new change. 

### How to test this PR?

Pipeline should be ✅

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -196,6 +196,7 @@
   "resolutions": {
     "trim": "0.0.3",
     "ssh2": "1.11.0",
-    "@types/node": "^20"
+    "@types/node": "^20",
+    "@types/docker-modem": "3.0.6"
```
Comment (benoitf) [inline package.json]: we don't need this resolution, you can just have the yarn.lock change

File: yarn.lock [hunk 2]
```diff
@@ -3857,18 +3857,18 @@
   dependencies:
     "@types/ms" "*"
 
-"@types/docker-modem@*":
-  version "3.0.2"
-  resolved "https://registry.npmjs.org/@types/docker-modem/-/docker-modem-3.0.2.tgz"
-  integrity sha512-qC7prjoEYR2QEe6SmCVfB1x3rfcQtUr1n4x89+3e0wSTMQ/KYCyf+/RAA9n2tllkkNc6//JMUZePdFRiGIWfaQ==
+"@types/docker-modem@*", "@types/docker-modem@3.0.6":
```
Comment (benoitf) [inline yarn.lock]:
```suggestion
"@types/docker-modem@*":
```

File: package.json (hunk 1 above)
Comment (axel7083) [inline package.json]:
```suggestion
    "@types/node": "^20"
```

File: packages/main/src/plugin/container-registry.ts [hunk 3]
```diff
@@ -1132,7 +1132,7 @@ export class ContainerProviderRegistry {
     try {
       const authconfig = this.imageRegistry.getAuthconfigForImage(imageName);
       const matchingEngine = this.getMatchingEngineFromConnection(providerContainerConnectionInfo);
-      const pullStream = await matchingEngine.pull(imageName, {
+      const pullStream: NodeJS.ReadableStream = await matchingEngine.pull(imageName, {
```
Comment (benoitf) [inline packages/main/src/plugin/container-registry.ts]: can't we omit NodeJS.ReadableStream there ?

File: yarn.lock [hunk 4]
```diff
@@ -2114,7 +2114,7 @@
     "@docusaurus/theme-search-algolia" "3.2.1"
     "@docusaurus/types" "3.2.1"
 
-"@docusaurus/react-loadable@5.5.2":
+"@docusaurus/react-loadable@5.5.2", "react-loadable@npm:@docusaurus/react-loadable@5.5.2":
```
Comment (benoitf) [inline yarn.lock]: do you know why docusaurus imports are being changed ?

File: packages/main/src/plugin/container-registry.ts (hunk 3 above)
Comment (axel7083) [reply packages/main/src/plugin/container-registry.ts]: yes

**Review by benoitf — APPROVED** [review]
Thx

**Pull request merged**